.github/actions/install-rust/action.yml

```diff
@@ -85,11 +85,15 @@ runs:
         echo RUSTFLAGS="-D warnings" >> "$GITHUB_ENV"
         echo RUSTDOCFLAGS="-D warnings" >> "$GITHUB_ENV"
 
+        if ${{ inputs.with_cache }}; then
         cat >> "$GITHUB_ENV" <<EOF
         RUSTC_WRAPPER=sccache
-        CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
         EOF
+        fi
 
+        cat >> "$GITHUB_ENV" <<EOF
+        CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
```

Comment:
What does `CARGO_REGISTRIES_CRATES_IO_PROTOCOL` mean? and setting it to `sparse`?

Reply:
Oh, it's https://blog.rust-lang.org/inside-rust/2023/01/30/cargo-sparse-protocol.html isn't it? Cool! TIL!